Type parameter nullability mismatch when using `@Nullable T` with `? extends @Nullable T`

I am working on enabling JSpecify mode in Spring Framework, and so far the most frequent source of issues has been the handling of `? extends @Nullable T` versus `@Nullable of T`.

See for example this simple code sample:
```
Supplier<? extends @Nullable String> stringSupplier = () -> null;
```
Which generates the following error:
```
error: [NullAway] Cannot assign from type Supplier<@Nullable String> to type Supplier<? extends @Nullable String> due to mismatched nullability of type parameters
```

It looks like a value use case to me, if not, please let me know.I have also the same issue with nullable method reference.

I think this might be another example of the same:

```java
@NullMarked
class Foo {
    static abstract class MyClass<T extends @Nullable Object>  {
        abstract T doThing(T value);
    }

    static void repro() {
      new MyClass<@Nullable Object>() {
        @Override
        @Nullable Object doThing(@Nullable Object value) {
            return value;
        }
      }.doThing(null);
    }
}
```

```
Foo.java:14: warning: [NullAway] method returns @Nullable, but superclass method Foo.MyClass.doThing(T) returns @NonNull
        @Nullable Object doThing(@Nullable Object value) {
                         ^
    (see http://t.uber.com/nullaway )
```

> I think this might be another example of the same:
> 
> @NullMarked
> class Foo {
>     static abstract class MyClass<T extends @Nullable Object>  {
>         abstract T doThing(T value);
>     }
> 
>     static void repro() {
>       new MyClass<@Nullable Object>() {
>         @Override
>         @Nullable Object doThing(@Nullable Object value) {
>             return value;
>         }
>       }.doThing(null);
>     }
> }
>  ```
> Foo.java:14: warning: [NullAway] method returns @Nullable, but superclass method Foo.MyClass.doThing(T) returns @NonNull
>         @Nullable Object doThing(@Nullable Object value) {
>                          ^
>     (see http://t.uber.com/nullaway )
> ```

@agrieve do you get this error with JSpecify mode enabled?  Without JSpecify mode it is expected as NullAway treats all type variables as `@NonNull`.  In any case I think it's a different issue than the one above.

--- a/nullaway/src/main/java/com/uber/nullaway/generics/CheckIdenticalNullabilityVisitor.java
+++ b/nullaway/src/main/java/com/uber/nullaway/generics/CheckIdenticalNullabilityVisitor.java
@@ -55,6 +55,11 @@ public class CheckIdenticalNullabilityVisitor extends Types.DefaultTypeVisitor<B
     for (int i = 0; i < lhsTypeArguments.size(); i++) {
       Type lhsTypeArgument = lhsTypeArguments.get(i);
       Type rhsTypeArgument = rhsTypeArguments.get(i);
+      if (lhsTypeArgument.getKind().equals(TypeKind.WILDCARD)
+          || rhsTypeArgument.getKind().equals(TypeKind.WILDCARD)) {
+        // TODO Handle wildcard types
+        continue;
+      }
       boolean isLHSNullableAnnotated = GenericsChecks.isNullableAnnotated(lhsTypeArgument, state);
       boolean isRHSNullableAnnotated = GenericsChecks.isNullableAnnotated(rhsTypeArgument, state);
       if (isLHSNullableAnnotated != isRHSNullableAnnotated) {

--- a/nullaway/src/test/java/com/uber/nullaway/jspecify/GenericsTests.java
+++ b/nullaway/src/test/java/com/uber/nullaway/jspecify/GenericsTests.java
@@ -2084,6 +2084,24 @@ public class GenericsTests extends NullAwayTestsBase {
             "        Supplier<? extends @Nullable Object> s = () -> null;",
             "    }",
             "}")
+        .addSourceLines(
+            "Test2.java",
+            "package com.uber;",
+            "import java.util.HashMap;",
+            "import java.util.Map;",
+            "import org.jspecify.annotations.Nullable;",
+            "import org.jetbrains.annotations.Contract;",
+            "public class Test2 {",
+            "    @Contract(\"null -> true\")",
+            "    public static boolean isEmpty(@Nullable Map<?, ? extends @Nullable Object> map) {",
+            "        return (map == null || map.isEmpty());",
+            "    }",
+            "    static void foo() {",
+            "        Map<String, @Nullable Object> variables = new HashMap<>();",
+            "        if (isEmpty(variables)) { /* do nothing */ }",
+            "        variables.toString();",
+            "    }",
+            "}")
         .doTest();
   }
 

--- a/nullaway/src/test/java/com/uber/nullaway/jspecify/GenericsTests.java
+++ b/nullaway/src/test/java/com/uber/nullaway/jspecify/GenericsTests.java
@@ -2069,6 +2069,24 @@ public class GenericsTests extends NullAwayTestsBase {
         .doTest();
   }
 
+  @Test
+  public void issue1126() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "import java.util.function.Supplier;",
+            "public class Test {",
+            "    static class K<T extends @Nullable Object> {}",
+            "    void foo(K<@Nullable Object> k) {",
+            "        K<? extends @Nullable Object> k2 = k;",
+            "        Supplier<? extends @Nullable Object> s = () -> null;",
+            "    }",
+            "}")
+        .doTest();
+  }
+
   private CompilationTestHelper makeHelper() {
     return makeTestHelperWithArgs(
         Arrays.asList(

